packages/agent-utils/src/llm/Chat.ts

```diff
@@ -124,28 +131,35 @@ export class Chat {
   }
 
   public async fitToContextWindow(): Promise<void> {
-    const msgLogs = this._msgLogs;
-    const totalTokens = () =>
-      msgLogs["persistent"].tokens +
-      msgLogs["temporary"].tokens;
-
-    if (totalTokens() < this._maxContextTokens) {
+    if (this.tokens < this._maxContextTokens) {
       return;
     }
 
-    this._logger.error(`! Max Tokens Exceeded (${totalTokens()} / ${this._maxContextTokens})`);
+    this._logger.error(`! Max Tokens Exceeded (${this.tokens} / ${this._maxContextTokens})`);
 
     // Start with "temporary" messages
     await this._summarize("temporary");
 
-    if (totalTokens() < this._maxContextTokens) {
+    if (this.tokens < this._maxContextTokens) {
       return;
     }
 
     // Move onto "persistent" messages
     await this._summarize("persistent");
   }
 
+  public export(): ChatMessageLog {
+    return JSON.parse(JSON.stringify(this._msgLogs));
```

Comment:
Nice deep clone, lmao

Reply:
lol yah, was testing before doing this and object references were getting updated after the initial "write" to the debug log.